### ISSUE TITLE
JASP crashes for mixed bayesian anova

### ISSUE THREAD

**Issue by metamorphose42** [jasp-stats/jasp-desktop#1807]
## Environment (for bugs)

* JASP version: 0.8.1.2
* OS name and version: macOS Sierra 10.12.4
* Analysis: Bayesian repeated measures ANOVA (mixed design)
## Environment (for enhancement requests)
Hi,
I'm trying to run a bayesian mixed design ANOVA (within:2x2x2 &between:x2x2). It works for just the within but when I add the between subject factors it doesn't compute (see screenshot for how I input my data) 

and then shuts JASP down. It gave me the message to let you guys know (see screenshot).
Thanks!
* Enhancement:
* Purpose:

<img width="1075" alt="screen shot 2017-06-10 at 15 29 52" src="https://user-images.githubusercontent.com/18419722/27003152-beb1701a-4df1-11e7-8255-4efec970aee9.png">

<img width="1389" alt="screen shot 2017-06-10 at 15 22 19" src="https://user-images.githubusercontent.com/18419722/27003096-e7f16d5a-4df0-11e7-963a-7eee94006ec1.png">

**Comment by boutinb** [comment]
Hi @metamorphose42, Can you send us the data & jasp file you use (in  a zip format)?

**Comment by metamorphose42** [comment]
Of course, thanks so much! Attached here.
[Archive.zip](https://github.com/jasp-stats/jasp-desktop/files/1067625/Archive.zip)

**Comment by boutinb** [comment]
@metamorphose42 We could reproduce this error. Hopefully we can solve this for the next release.

**Comment by metamorphose42** [comment]
So for now I cant run it or is there a way around it?

Thanks!

**Comment by TimKDJ** [comment]
The analysis you are trying to run would generate 7580 different models.
I had a look and this appears to cause a capacity problem:
`get_next_capacity, allocator's max_size reached`

You could perform a more targeted analysis by specifying terms you are not directly interested in as nuisance.
See also http://www.ejwagenmakers.com/inpress/RouderEtAlinpressANOVAPM.pdf (p. 24-26) for strategies for trimming down the model space.

Hopefully we can solve this in the next release!

**Comment by metamorphose42** [comment]
Thanks so much!

I thought of trimming, but to look at interactions only I still need to include the factors by themselves in the analysis, right?

I will think more and look forward to the next release, thanks so much for all your help and amazing work on JASP, it's truely appreciatd!

Best,
Clarisse

**Comment by TimKDJ** [comment]
Thank you for your kind words!

The factors do indeed need to be included, but...

> Consider the variation across participants and items. This variation certainly needs to be modeled, but it need not be tested. Hence, all models may include variation from these factors, and models without this variation need not be considered.

(from the article I linked)

You can specify main effects as nuisance and still test their interaction.

**Comment by metamorphose42** [comment]
Amazing, thanks so much for clarifying!

Best,
Clarisse

**Comment by metamorphose42** [comment]
Hi,

I have now update to version 0.8.4, and trying to run my 7580 models... I don't get an error message, but I'm not sure how to tell whether it's working (how long should I wait?) or whether Jasp keeps crashing (its still responsive, i.e. I can move around within the interface, just no BFs are coming up)?

Here a screenshot:

<img width="1212" alt="screen shot 2017-12-19 at 09 39 09" src="https://user-images.githubusercontent.com/18419722/34148174-942c6944-e4a0-11e7-80d8-9e6832294529.png">

Thanks!

Best,
Clarisse

**Comment by EJWagenmakers** [comment]
You can see the progress bar on top, and it is not looking like the analysis will end any time soon. Have you tried running the analyses with fewer predictors first? Also, Tim said that he could "hopefully" solve it in the next release, but I'm not sure he got to it. @FransMeerhoff

**Comment by metamorphose42** [comment]
Yes, I have tried, it has always worked for just within subject factors, when it comes to between it stops working. And I tried specifying some as nuissance, but it still crashes, maybe because I have to enter my data before I can select which are nuissance, so it already gets started?

**Comment by EJWagenmakers** [comment]
If you send me the data set (or part of it), I can take a look.
EJ.Wagenmakers@gmail.com

**Comment by metamorphose42** [comment]
That would be great, thank you so much! csv file attached.

What I am hoping to redo as in the frequentist analyses is this:

Within: (2x2x2)
T1accHH [1, 1, 1]
T1accHL [1, 1, 2]
T1accLH [1, 2, 1]
T1accLL [1, 2, 2]
T2accHH [2, 1, 1]
T2accLH [2, 1, 2]
T2accHL [2, 2, 1]
T2accLL [2, 2, 2]

Between:
GainLoss
SelfOther

Thank you!

**Comment by AlexanderLyNL** [comment]
The analysis is still working, but it takes a while before it's able to enumerate all models, which takes a very long time.

**Comment by EJWagenmakers** [comment]
So then the suggestion would be to start the analysis in the evening, and by the morning you should at least see from the progress bar how much longer it will take...
E.J.

**Comment by metamorphose42** [comment]
Thank you both, I will do that and look forward to the morning :)

**Comment by AlexanderLyNL** [comment]
Sorry for this, but this is a hard problem, because the number of models get multiplied by each level in the factor and therefore increasing the number of models very rapidly. The increased number of models also test the limits of memory and the pipeline between our front-end and back-end. We try to be thorough here and provide Bayes factors for all the models, but this might actually be an overkill.
